safety_shield/scripts/urdf_capsule_generation.py

```diff
@@ -0,0 +1,527 @@
+import os
+import math
+import re
+import warnings
+
+import numpy as np
+import matplotlib.pyplot as plt
+import trimesh
+import xml.etree.ElementTree as ET
+from sklearn.decomposition import PCA
+
+# make console output more readable
+np.set_printoptions(threshold=np.inf)
+np.set_printoptions(suppress=True)
+
+
+def export_PC(point_cloud, name):
+    """
+    debug function: export point cloud to ply file (can be imported to blender)
+    :param point_cloud: the point cloud to export
+    :param name: name of the file
+    :return: void
+    """
+    print("saving to file: " + name)
+    with open(name + '.ply', "w") as file:
+        # HEADER
+        file.write("ply\n")
+        file.write("format ascii 1.0\n")
+        file.write("element vertex {}\n".format(point_cloud.shape[0]))
+        file.write("property float x\n")
+        file.write("property float y\n")
+        file.write("property float z\n")
+        file.write("end_header\n")
+
+        # write points
+        for p in point_cloud:
+            file.write(
+                "{} {} {} \n".format(p[0], p[1], p[2]))
+
+
+def visualize_meshes(*meshes):
+    """
+    visualize tri meshes for debugging purposes
+    :param meshes: all meshes that should be visualized in same plot
+    :return: void
+    """
+    fig = plt.figure()
+    ax = fig.add_subplot(projection='3d')
+
+    for mesh in meshes:
+        ax.plot_trisurf(mesh.vertices[:, 0], mesh.vertices[:, 1], mesh.vertices[:, 2], triangles=mesh.faces)
+
+
+def rpy_to_rot_matrix(rpy):
+    """
+    construct rotation matrix from RPY parameters
+    :param rpy: np array with roll, pitch and yaw
+    :return: 3x3 array representing the rotation matrix
+    """
+    roll = rpy[0]
+    pitch = rpy[1]
+    yaw = rpy[2]
+
+    Rx = np.array([[1, 0, 0],
+                   [0, np.cos(roll), -np.sin(roll)],
+                   [0, np.sin(roll), np.cos(roll)]])
+
+    Ry = np.array([[np.cos(pitch), 0, -np.sin(pitch)],  # correct?
+                   [0, 1, 0],
+                   [np.sin(pitch), 0, np.cos(pitch)]])  # correct?
+
+    Rz = np.array([[np.cos(yaw), -np.sin(yaw), 0],
+                   [np.sin(yaw), np.cos(yaw), 0],
+                   [0, 0, 1]])
+
+    R = Rz @ Ry @ Rx
+    return R
+
+
+def get_mesh_filename(link):
```

Comment:
either rename to `get_mesh_filename_and_scale` or add a docstring.

Reply:
Both added in 9d4aa102acc213e435fb6567d7b610de0458b94d